app/src/main/java/com/github/swent/echo/authentication/AuthenticationService.kt

```diff
@@ -36,7 +39,19 @@ interface AuthenticationService {
     /**
      * Get the current user's ID.
      *
+     * Should be called after [initialize], otherwise might return null even if there exists an
+     * authenticated user.
+     *
      * @return The current user's ID, or null if the user is not signed in.
      */
-    suspend fun getCurrentUserID(): String?
+    fun getCurrentUserID(): String?
+
+    /**
+     * Check if the user is logged in.
+     *
+     * @return True if the user is logged in, false otherwise.
+     */
+    fun userIsLoggedIn(): Boolean {
```

Comment:
Then this could be a stateflow to just recompose the AppNavigationHost when the user is logged in